### PR TITLE
wxGTK30-gtk2: migrate most packages to wxGTK{32,30-gtk3}

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mgttlinger — CHANGES_REQUESTED** [review]
As maintainer of the urbackup client, I tested that and it seems to work, however several other packages failed to build as result of this PR:
```
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/198552

10 packages failed to build:
perl534Packages.AlienWxWidgets perl534Packages.AppMusicChordPro perl534Packages.Wx perl534Packages.WxGLCanvas perl536Packages.AlienWxWidgets perl536Packages.AppMusicChordPro perl536Packages.Wx perl536Packages.WxGLCanvas slic3r springLobby
```

**Comment by wegank** [comment]
@mgttlinger Removed the Perl commit. However, I don't have `springLobby` failing on my side. Could you please test again?

**Comment by mgttlinger** [comment]
> @mgttlinger Removed the Perl commit. However, I don't have `springLobby` failing on my side. Could you please test again?

Yes, sadly it still fails:
```
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/198552

1 package failed to build:
springLobby

14 packages built:
boinc dolphin-emu electricsheep fityk flamerobin lenmus lutris lutris-free odamex pwsafe sooperlooper urbackup-client xchm xylib
```

I'm using `nixpkgs-review` to review this.

**Comment by wegank** [comment]
@mgttlinger Alright. Removed springLobby from the PR.

**Comment by mgttlinger** [comment]
Now there is only a broken package:
```
1 package marked as broken and skipped:
dolphinEmu

14 packages built:
boinc dolphin-emu electricsheep fityk flamerobin lenmus lutris lutris-free odamex pwsafe sooperlooper urbackup-client xchm xylib
```

**Comment by wegank** [comment]
Wait, but `dolphinEmu = dolphin-emu;` in `pkgs/top-level/aliases.nix` ...

**Comment by wegank** [comment]
@ofborg build dolphinEmu

**Comment by wegank** [comment]
@mgttlinger I think it's more of a false alarm.

**Review by mgttlinger — CHANGES_REQUESTED** [review]
Otherwise, the changes look good to me.

File: pkgs/games/odamex/default.nix [hunk 1]
```diff
@@ -9,14 +19,36 @@ stdenv.mkDerivation rec {
...
+  '') + ''
+    runHook postInstall
+  '';
 
   meta = {
     homepage = "http://odamex.net/";
     description = "A client/server port for playing old-school Doom online";
-    license = lib.licenses.gpl2;
-    platforms = lib.platforms.linux;
+    license = lib.licenses.gpl2Plus;
```
[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/games/odamex/default.nix]: Done.

File: pkgs/applications/science/misc/boinc/default.nix [hunk 2]
```diff
@@ -45,11 +81,7 @@ stdenv.mkDerivation rec {
     description = "Free software for distributed and grid computing";
     homepage = "https://boinc.berkeley.edu/";
     license = licenses.lgpl2Plus;
-    platforms = platforms.linux;  # arbitrary choice
-    # checking for gcc options needed to detect all undeclared functions... cannot detect
-    # configure: error: in `/build/boinc-7.18.1-src':
-    # configure: error: cannot make gcc report undeclared builtins
-    broken = stdenv.isAarch64;
+    platforms = platforms.linux; # arbitrary choice
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/boinc/default.nix:84]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/games/spring/default.nix [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+, SDL2
+, libGLU
+, libGL
+, asciidoc
+, docbook_xsl
+, docbook_xsl_ns
+, curl
+, makeWrapper
+, jdk ? null
+, python ? null
```
Comment (SuperSandro2000) [inline pkgs/games/spring/default.nix:24]:
```suggestion
, jdk
, python
```

**Review by SuperSandro2000 — COMMENTED** [review]
due to all the formatting changes it is not easy to know the actual changes. I didn't notice anything else though.

**Comment by Luflosi** [comment]
The changes to BOINC LGTM.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 198552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>boinc</li>
    <li>dolphin-emu</li>
    <li>electricsheep</li>
    <li>fityk</li>
    <li>flamerobin</li>
    <li>lenmus</li>
    <li>lutris</li>
    <li>lutris-free</li>
    <li>odamex</li>
    <li>pwsafe</li>
    <li>sooperlooper</li>
    <li>urbackup-client</li>
    <li>xchm</li>
    <li>xylib</li>
  </ul>
</details>

**Pull request merged**